File: DSA/VikrantSinghBanwala_590018082/day76_question1.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int data;
    struct Node *left, *right;
} Node;

Node* newNode(int data) {
    Node* node = (Node*)malloc(sizeof(Node));
    node->data = data;
    node->left = node->right = NULL;
    return node;
}

int toSumTree(Node* root) {
    if (!root) return 0;
    int leftSum = toSumTree(root->left);
    int rightSum = toSumTree(root->right);
    int oldVal = root->data;
    root->data = leftSum + rightSum;
    return root->data + oldVal;
}

void printInOrder(Node* root) {
    if (!root) return;
    printInOrder(root->left);
    printf("%d ", root->data);
    printInOrder(root->right);
}

int main() {
    Node* root = newNode(10);
    root->left = newNode(-2);
    root->right = newNode(6);
    root->left->left = newNode(8);
    root->left->right = newNode(-4);
    root->right->left = newNode(7);
    root->right->right = newNode(5);

    toSumTree(root);

    printInOrder(root);
    return 0;
}

File: DSA/akashyadav_590014126/Day_76_Question_1.py
class Node:
    def __init__(self, val):
        self.data = val
        self.left = None
        self.right = None

def convert_to_sum_tree(root):
    def post_order(node):
        if node is None:
            return 0

        left_sum = post_order(node.left)
        right_sum = post_order(node.right)

        old_val = node.data

        node.data = left_sum + right_sum

        return node.data + old_val

    post_order(root)
